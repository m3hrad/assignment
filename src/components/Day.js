import React from 'react';
import PropTypes from 'prop-types';
let moment = require('moment');


export default class Day extends React.Component {
  isEven(value){
    return (value % 2) === 0;
  }

  capitalizeFirstLetter(string) {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }

  render() {
    const today = moment().format('dddd');
    const capitalizedDay = this.capitalizeFirstLetter(this.props.day);
    let todayComponent = '';
    if (capitalizedDay === today) {
      todayComponent = <span>TODAY</span>;

    }


    let displayedHours = '';
    let hours = this.props.hours;

    if (hours === undefined || hours.length === 0) {
      displayedHours = 'Closed';
    }
    else {
      displayedHours = this.props.hours.map((hour,index) => {
        if (hour.type === 'close'){
          hour.value = ' - ' + hour.value;
        }
        if (this.isEven(index) && index !== 0){
          hour.value = ', ' + hour.value;
        }
        return hour.value;
      });
    }

    return (
      <div>
        {capitalizedDay}
        {todayComponent}
        {displayedHours}
      </div>
    );
  }
}

Day.propTypes = {
  day: PropTypes.any.isRequired,
  hours: PropTypes.any.isRequired,
};