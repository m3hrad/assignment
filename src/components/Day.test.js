import React from 'react';
import {shallow} from 'enzyme';
import Day from './Day';

const day = require('../input/day1');
const key = 'Monday';

describe('Day ', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Day key={key} day={key} hours={day[key]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});