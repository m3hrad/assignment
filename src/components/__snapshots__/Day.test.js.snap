// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Day  renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Day
    day="Monday"
    hours={
      Array [
        Object {
          "type": "open",
          "value": "12 PM",
        },
        Object {
          "type": "close",
          "value": " -  - 8 PM",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          style={
            Object {
              "minWidth": "140px",
            }
          }
        >
          Monday
          
        </span>,
        <span
          style={
            Object {
              "color": "rgba(32,33,37,1)",
              "fontFamily": "Circular Std Book",
            }
          }
        >
          12 PM
           -  - 8 PM
        </span>,
      ],
      "style": Object {
        "borderBottom": "1px solid rgba(32,33,37,0.08)",
        "display": "flex",
        "fontSize": "16px",
        "justifyContent": "space-between",
        "lineHeight": "22px",
        "paddingBottom": "7px",
        "paddingTop": "7px",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Monday",
            "",
          ],
          "style": Object {
            "minWidth": "140px",
          },
        },
        "ref": null,
        "rendered": Array [
          "Monday",
          "",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "12 PM",
            " -  - 8 PM",
          ],
          "style": Object {
            "color": "rgba(32,33,37,1)",
            "fontFamily": "Circular Std Book",
          },
        },
        "ref": null,
        "rendered": Array [
          "12 PM",
          " -  - 8 PM",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            style={
              Object {
                "minWidth": "140px",
              }
            }
          >
            Monday
            
          </span>,
          <span
            style={
              Object {
                "color": "rgba(32,33,37,1)",
                "fontFamily": "Circular Std Book",
              }
            }
          >
            12 PM
             -  - 8 PM
          </span>,
        ],
        "style": Object {
          "borderBottom": "1px solid rgba(32,33,37,0.08)",
          "display": "flex",
          "fontSize": "16px",
          "justifyContent": "space-between",
          "lineHeight": "22px",
          "paddingBottom": "7px",
          "paddingTop": "7px",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Monday",
              "",
            ],
            "style": Object {
              "minWidth": "140px",
            },
          },
          "ref": null,
          "rendered": Array [
            "Monday",
            "",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "12 PM",
              " -  - 8 PM",
            ],
            "style": Object {
              "color": "rgba(32,33,37,1)",
              "fontFamily": "Circular Std Book",
            },
          },
          "ref": null,
          "rendered": Array [
            "12 PM",
            " -  - 8 PM",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
